File: VB/02_Text/GetTextSizeBasedOnFont.vb
Imports System.IO
Imports System.Text
Imports Spire.Pdf.Graphics

Namespace GetTextSizeBasedOnFont
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a StringBuilder instance to store the text size details
			Dim stringB As New StringBuilder()

			' Specify the text to measure
			Dim text As String = "Spire.PDF for .NET"

			' Create an instance of PdfFont using the Times Roman font family and size 12
			Dim font1 As New PdfFont(PdfFontFamily.TimesRoman, 12.0F)

			' Measure the size of the text based on the font
			Dim sizeF1 As SizeF = font1.MeasureString(text)

			' Append the details of the text size to the StringBuilder
			stringB.AppendLine("1. The width is: " & sizeF1.Width & ", the height is: " & sizeF1.Height)

			' Create an instance of PdfTrueTypeFont using the Arial font, size 12, and regular style
			Dim font2 As New PdfTrueTypeFont(New Font("Arial", 12.0F, FontStyle.Regular), True)

			' =============================================================================
			' Use the following code for netstandard dlls
			' =============================================================================
			' Dim font2 As New PdfTrueTypeFont("Arial", 12.0F, FontStyle.Regular, True)
			' =============================================================================

			' Measure the size of the text based on the font
			Dim sizeF2 As SizeF = font2.MeasureString(text)

			' Append the details of the text size to the StringBuilder
			stringB.AppendLine("2. The width is: " & sizeF2.Width & ", the height is: " & sizeF2.Height)

			' Specify the output file name
			Dim result As String = "GetTextSizeBasedOnFont_out.txt"

			' Write the content of the StringBuilder to a text file
			File.WriteAllText(result, stringB.ToString())

			'Launch the Txt file
			PDFDocumentViewer(result)
		End Sub

		Private Sub PDFDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

	End Class
End Namespace

File: CS/02_Text/GetTextSizeBasedOnFont.cs
using System;
using System.Drawing;
using System.IO;
using System.Text;
using System.Windows.Forms;
using Spire.Pdf.Graphics;


namespace GetTextSizeBasedOnFont
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            StringBuilder stringB = new StringBuilder();
            string text = "Spire.PDF for .NET";

            //Create an instance for PdfFont
            PdfFont font1 = new PdfFont(PdfFontFamily.TimesRoman, 12f);

            //Get text size based on font name and size
            SizeF sizeF1 = font1.MeasureString(text);
            stringB.AppendLine("1. The width is: " + sizeF1.Width + ", the height is: " + sizeF1.Height);

            //Create an instance for PdfTrueTypeFont
            PdfTrueTypeFont font2 = new PdfTrueTypeFont(new Font("Arial", 12f, FontStyle.Regular), true);

            //Get text size based on font name and size
            SizeF sizeF2 = font2.MeasureString(text);
            stringB.AppendLine("2. The width is: " + sizeF2.Width + ", the height is: " + sizeF2.Height);

            //Save the result txt file
            string result = "GetTextSizeBasedOnFont_out.txt";
            File.WriteAllText(result, stringB.ToString());

            //Launch the Txt file
            PDFDocumentViewer(result);
        }

        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}
